py3dtilers/ObjTiler/obj.py

```diff
@@ -110,22 +114,31 @@ def retrieve_objs(files, with_texture=False):
         Create Obj instance from OBJ file(s).
         :param files: paths to files
         :param with_texture: a boolean indicating if the textures should be read
-
         :return: a list of Obj.
         """
         objects = list()
 
         for obj_file in files:
             print("Reading " + str(obj_file))
-            geom = pywavefront.Wavefront(obj_file, collect_faces=True)
+            geom = pywavefront.Wavefront(obj_file, collect_faces=True, create_materials=True)
+            mesh = geom.mesh_list[0]
             if len(geom.vertices) == 0:
                 continue
-            for mesh in geom.mesh_list:
-                # Get id from its name
+            gltfMaterials = []
+            mesh_index = 1
+
+            for mesh in mesh.materials:
```

Comment:
We need to iterate on all meshes then all materials, so we need a double for loop:  

```python
            for mesh in geom.mesh_list:
                for mesh_mat in mesh.materials:
                    # get id from its name
                    id = mesh.name + str(mesh_index)
                    obj = Obj(id)
                    obj.set_material_index(mesh_index)
                    mesh_index += 1
                    if obj.parse_geom(mesh_mat, with_texture):
                        objects.append(obj)
                    material = GlTFMaterial(rgb=[mesh_mat.diffuse[0], mesh_mat.diffuse[1], mesh_mat.diffuse[2]], alpha=1. - mesh_mat.diffuse[3], metallicFactor=0.)
                    gltfMaterials.append(material)
```